core/commands/netdiag.go

```diff
@@ -0,0 +1,133 @@
+package commands
+
+import (
+	"fmt"
+	"io"
+	"text/template"
+
+	"github.com/ipfs/go-ipfs/core/commands/cmdenv"
+	"github.com/ipfs/go-ipfs/doctor"
+
+	cmds "github.com/ipfs/go-ipfs-cmds"
+
+	libp2pNetwork "github.com/libp2p/go-libp2p-core/network"
+)
+
+type NetworkDiagnostics struct {
+	doctor.Status
+	IsOnline bool
+}
+
+const netDiagPublicText = `
+You are online and reachable from the public network.
+`
+
+const netDiagUnknownText = `
+IPFS has not yet determined if your node is reachable from the public network.
+`
+
+const netDiagPrivateText = `
+{{ if .UsingRelays -}}
+Your IPFS node appears to be behind a Firewall or NAT and can only be reached
+through a relay.
+{{- else -}}
+{{if .AutoRelayEnabled -}}
+Your IPFS node appears to be behind a Firewall or NAT and will try to
+automatically find a relay. However, it is not currently reachable.
+{{- else -}}
+Your IPFS node appears to be behind a Firewall or NAT and cannot be reached by
+nodes in the public network.
+
+To join the network via a relay, please enable the "auto-relay" feature:
+
+    ipfs config --bool Swarm.EnableAutoRelay true 
+
+And then restart IPFS.
+{{- end -}}
+{{end}}
+
+Please note, relays are public and highly constrained resources.
+
+For the best performance, please enable Universal Plug n Play (UPnP) on your
+router so IPFS can automatically configure it to open an inbound port.
+
+{{ if or .TCPPorts .UDPPorts -}}
+Alternatively, you can manually configure your router. Please forward the following ports:
+
+{{ if .TCPPorts -}}
+TCP:{{ range $element := .TCPPorts}} {{$element}}{{end}}
+{{- end}}{{if .UDPPorts -}}
+UDP:{{ range $element := .UDPPorts}} {{$element}}{{end}}
+{{- end}}
+
+To (your current IP): {{.LocalIP}}
+{{- end}}
+{{if .Gateway -}}
+Your router's administrative console can likely be found at:
```

Comment:
this is less likely to be actionable in corporate / educational situations. Maybe clearer specification of "if you are on a personal home network" or similar?